sound/soc/sof/intel/hda-codec.c

```diff
@@ -263,6 +263,7 @@ int hda_codec_i915_exit(struct snd_sof_dev *sdev)
 }
 EXPORT_SYMBOL_NS(hda_codec_i915_exit, SND_SOC_SOF_HDA_AUDIO_CODEC_I915);
 
+MODULE_SOFTDEP("pre: snd-hda-codec-hdmi");
```

Comment:
In such environment I think this should be handled by a conf file in /etc/modprobe.d with content
```
options snd_sof_intel_hda softdep pre: snd-hda-codec-hdmi
```
Btw,  what does "In environment without KMOD requesting module may fail to load snd-hda-codec-hdmi" means in practice? What kind of environment is this refers to? Can you detail the issue?

Reply:
> In such environment I think this should be handled by a conf file in /etc/modprobe.d with content

I would argue this is more ideal. If I understand correctly, when `CONFIG_SND_HDA_CODEC_HDMI` is enabled, during init the driver should always try to probe `snd-hda-codec-hdmi` through `request_codec_module`. Adding `softdep` behind the gated part is correctly stating the soft dependency. Just my 2 cents.

Is there anything I am missing here that could break things?

> What kind of environment is this refers to? Can you detail the issue?

In Android environment, kernel module request is blocked as a whole by SELinux. i.e. `request_module` will always get blocked by SELinux.

Reply:
Fair enough.